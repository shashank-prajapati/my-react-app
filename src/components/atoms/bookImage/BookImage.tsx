import React from "react";

type Props = {
  source: string;
  width?: string | number | undefined
};

function BookImage(props: Props) {
  const { source,width } = props;
  return (
    <img
      style={{ display: "block" }}
      src={source}
      width={width}
      alt="Book Cover"
    ></img>
  );
}

export default BookImage;
