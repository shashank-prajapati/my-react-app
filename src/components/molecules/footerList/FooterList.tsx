import React from 'react';
import Typography from '@mui/material/Typography';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';

type Props = {
    list:Array<string>;
    heading:string;
};

const FooterList = (props: Props) => {
  const {list,heading} = props;

  const listItems = list.map(item=>(<ListItem><Typography variant='body2' color='extraTextColor.main'>{item}</Typography></ListItem>))
  return <div>
      <List>
          <ListItem>
          <Typography variant='body1' color='navTextColors' >{heading}</Typography>
          </ListItem>
          {listItems}
      </List>
  </div>;
}

export default FooterList;
