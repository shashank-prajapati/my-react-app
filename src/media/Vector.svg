<svg width="16" height="23" viewBox="0 0 16 23" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.502 19C10.8404 20.5823 9.58232 21.8404 8 22.502C6.41768 21.8404 5.15956 20.5823 4.498 19H6.758C7.084 19.489 7.505 19.912 8 20.243C8.495 19.913 8.916 19.489 9.243 19H11.502ZM14 13.805L16 16.073V18H0V16.073L2 13.805V7.99996C2 4.51696 4.504 1.55296 8 0.454956C11.496 1.55296 14 4.51696 14 7.99996V13.805ZM13.27 16L12 14.56V7.99996C12 5.68196 10.43 3.56996 8 2.57996C5.57 3.56996 4 5.68096 4 7.99996V14.56L2.73 16H13.27ZM8 9.99996C7.46957 9.99996 6.96086 9.78924 6.58579 9.41417C6.21071 9.0391 6 8.53039 6 7.99996C6 7.46952 6.21071 6.96082 6.58579 6.58574C6.96086 6.21067 7.46957 5.99996 8 5.99996C8.53043 5.99996 9.03914 6.21067 9.41421 6.58574C9.78929 6.96082 10 7.46952 10 7.99996C10 8.53039 9.78929 9.0391 9.41421 9.41417C9.03914 9.78924 8.53043 9.99996 8 9.99996Z"/>
</svg>
