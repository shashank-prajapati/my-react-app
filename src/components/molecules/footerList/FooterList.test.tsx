import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import FooterList from "./FooterList";
import { Typography } from "@mui/material";
import BookImage from "../../../media/coverImage.png";
import { ReactComponent as EntrepreneurshipIcon } from "../../../media/Vector.svg";

describe("Testing Footer List", () => {
  const list1 = [
    "Book lists",
    "What is Nonfiction?",
    "What to read next?",
    "Benefits of reading",
  ];
  it("should match the Footer List snapshot", () => {
    const { container } = render(
      <FooterList list={list1} heading="Editorial" />
    );
    expect(container).toMatchSnapshot();
  });
});
