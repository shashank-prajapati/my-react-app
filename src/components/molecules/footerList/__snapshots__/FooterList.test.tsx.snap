// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer List should match the Footer List snapshot 1`] = `
<div>
  <div>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1cka7m9-MuiTypography-root"
        >
          Editorial
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
        >
          Book lists
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
        >
          What is Nonfiction?
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
        >
          What to read next?
        </p>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1pjftnz-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </li>
    </ul>
  </div>
</div>
`;
